orchagent/pfcactionhandler.cpp

```diff
@@ -55,14 +55,23 @@ void PfcWdActionHandler::initCounters(void)
     }
 
     auto wdQueueStats = getQueueStats(m_countersTable, sai_serialize_object_id(m_queue));
-    wdQueueStats.detectCount++;
+    // initCounters() is called when the event channel receives
+    // a storm signal. This can happen when there is a true new storm or
+    // when there is an existing storm ongoing before warm-reboot. In the latter case,
+    // we treat the storm as an old storm. In particular,
+    // we do not increment the detectCount so as to clutch the
+    // gap between detectCount and restoreCount by 1 at maximum
+    if (!(wdQueueStats.detectCount > wdQueueStats.restoreCount))
```

Comment:
>if (!(wdQueueStats.detectCount > wdQueueStats.restoreCount)) [](start = 4, length = 60)

The comment said "when there is an existing storm ongoing before warm-reboot", however this if-condition seems not related.

Reply:
"when there is an existing storm ongoing before warm-reboot", It is indicated in the counters, pfcwd show stat would have STORM DETECTED = STORM RESTORED + 1, the values of which are carried over to post warm-reboot.

       QUEUE    STORM DETECTED/RESTORED    TX OK/DROP    RX OK/DROP    TX LAST OK/DROP    RX LAST OK/DROP
------------  -------------------------  ------------  ------------  -----------------  -----------------
Ethernet36:3                        2/1         0/300         0/300              0/100              0/100
Ethernet36:4                        2/1         0/300         0/300              0/100              0/100
Ethernet96:3                        2/1         0/300         0/300              0/100              0/100
Ethernet96:4                        2/1         0/300         0/300              0/100              0/100